packages/bee-q/src/components/notification/bq-notification.tsx

```diff
@@ -0,0 +1,188 @@
+  // =======================
+  @Watch('type')
+  checkPropValues() {
+    validatePropValue(NOTIFICATION_TYPES, 'default', this.el, 'type');
+  }
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentWillLoad() {
+    this.checkPropValues();
+  }
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  /**
+   * Trigger function when you want to close Notification
+   */
+  @Method()
+  async hide() {
+    this.hideNotification();
+  }
+
+  /**
+   * Trigger function when you want to show Notification
+   */
+  @Method()
+  async show() {
+    this.showNotification();
+  }
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private getNotificationIconName = () => {
+    if (!this.showIcon) return '';
+
+    const type = this.type;
+    let icon = 'info';
+
+    if (type === 'success') {
+      icon = 'check-circle';
+    }
+
+    if (type === 'error') {
+      icon = 'x-circle';
+    }
+
+    return icon;
+  };
+
+  private getNotificationColor = () => {
+    const type = this.type;
+    const colors = {
+      default: 'ui--inverse-active',
+      success: 'ui--success',
+      info: 'ui--brand',
+      warning: 'ui--warning',
+      error: 'ui--danger',
+    };
+
+    return isDefined(this.subjectColor) ? this.subjectColor : colors[type];
+  };
+
+  private hideNotification = () => {
+    this.isHidden = true;
+  };
+
+  private showNotification = () => {
+    this.isHidden = false;
+  };
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    const styles = {
+      ...(this.subjectColor && { color: getColorCSSVariable(this.subjectColor) }),
+    };
+
+    return (
+      <Host style={styles} aria-hidden={this.isHidden} hidden={this.isHidden}>
+        <div
+          class="bg-white notification-shadow notification-radius inline-block w-auto min-w-[var(--bq-notification--min-width)] p-[var(--bq-notification--padding)]"
+          part="base"
+        >
+          {this.showIcon && (
+            <div class="mr-2 inline-block text-left align-top" part="icon">
+              <bq-icon name={this.getNotificationIconName()} color={this.getNotificationColor()} size="24"></bq-icon>
+            </div>
+          )}
+          {!this.showIcon && (
+            <div class="avatar-slot-holder inline-block text-left align-top" part="avatar">
+              <slot name="avatar" />
+            </div>
+          )}
```

Comment:
I don't think we need a separate slot for avatar and icon, usually, if the consumer would like to show an avatar instead of an icon, the icon slot should be the one to use since the intention is the same. That said, maybe something like this could work 👇🏼 

```jsx
{/* ICON */}
{!this.hideIcon && (
  <div
    class={{
      'notification--icon mr-2 inline-block text-left align-top': true,
      [`color-${this.type}`]: true, // The icon color will be based on the type (info, success, warning, error)
    }}
  >
    {!this.hasCustomIcon ? <bq-icon name={this.iconName} /> : <slot name="icon" />}
  </div>
)}
```

By adding a property `hasCustomIcon` we can provide the flexibility to use the icon slot to display something different from the predefined icons, even an avatar.